Geometry/HcalCommonData/src/HcalDDDRecConstants.cc

```diff
@@ -350,19 +362,20 @@ void HcalDDDRecConstants::initialize(void) {
 					  << " at index " << i 
 					  << " of etaTable from SimConstant";
     } else {
+      etaTable.push_back(hpar->etaTable[ieta]);
       etaSimValu.push_back(std::pair<int,int>(ef,ieta));
     }
-    for (int k=0; k<hpar->etagroup[i]; ++k) ietaMap.push_back(i+1);
+    for (int k=0; k<(hpar->etagroup[i]); ++k) ietaMap.push_back(i+1);
     if (ieta <= hpar->etaMax[0]) ietaHB = i+1;
     if (ieta <= hpar->etaMin[1]) ietaHE = i+1;
   }
-  assert( hpar->etaMin[1] == ietaHE );
-  assert( hpar->etaMax[0] == ietaHB );
+  iEtaMin[1] = ietaHE;
```

Comment:
When we use etagouping it will change - both etaTable as well as etaMax/Min. This does not happen for Phase0/PhaseI but in PhaseII. I can cross check PhaseII scenario later
________________________________
From: Ianna Osborne [notifications@github.com]
Sent: 17 July 2015 12:47
To: cms-sw/cmssw
Cc: Sunanda Banerjee
Subject: Re: [cmssw] bsunanda:Run2-hcx22 Adopt simulation code to the new version of Hcal geometry (#10247)


In Geometry/HcalCommonData/src/HcalDDDRecConstants.cc<https://github.com/cms-sw/cmssw/pull/10247#discussion_r34879658>:

>      if (ieta <= hpar->etaMax[0]) ietaHB = i+1;
>      if (ieta <= hpar->etaMin[1]) ietaHE = i+1;
>    }
> -  assert( hpar->etaMin[1] == ietaHE );
> -  assert( hpar->etaMax[0] == ietaHB );
> +  iEtaMin[1] = ietaHE;


@bsunanda<https://github.com/bsunanda> - according to your code it is not possible - both iEtaMax and iEtaMin are copies of hpar->etaMin and hpar->etaMax (see lines 348 and 349). They are not modified after that assignment.

—
Reply to this email directly or view it on GitHub<https://github.com/cms-sw/cmssw/pull/10247/files#r34879658>.